File: Data Structures & Algorithms/two-integer-sum/submission-12.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        dictionary = {}
        for i in range(len(nums)):
            diff = target - nums[i]
            if diff in dictionary:
                return [dictionary[diff], i]
            dictionary[nums[i]] = i
        return []

File: Data Structures & Algorithms/two-integer-sum-ii/submission-12.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int l = 0, r = numbers.size() - 1;
        while (l < r) {
            int currSum = numbers.at(l) + numbers.at(r);
            if (currSum == target) {
                return {l + 1, r + 1};
            } else if(currSum < target) {
                ++l;
            } else {
                --r;
            }
        }
        return {0, 0};

    }
};
